CesiumAsync/include/CesiumAsync/Future.h

```diff
@@ -175,18 +174,10 @@ template <typename T> class Future final {
    * deadlock because the main thread tasks will never complete while this
    * method is blocking the main thread.
    *
-   * @return The value if the future resolves successfully, or the exception if
-   * it rejects.
+   * @return The value if the future resolves successfully.
+   * @throws An exception if the future rejected.
    */
-  Impl::FutureWaitResult_t<T> wait() {
-    try {
-      return Impl::FutureWaitResult<T>::getFromTask(this->_task);
-    } catch (std::exception& e) {
-      return e;
-    } catch (...) {
-      return std::runtime_error("Unknown exception.");
-    }
-  }
+  T wait() { return this->_task.get(); }
```

Comment:
I had such an interface originally. It would return `std::variant<T, std::exception>`. But that didn't work at all, because copying the exception into the variant would slice the exception (i.e. call the std::exception copy/move constructor on an instance that is actually of a derived type). Once I realized that, I tried making it `std::variant<T, std::exception_ptr>` instead, but that doesn't really help because the only way to get an exception from an exception_ptr is by throwing, so it doesn't help clients with exceptions disabled. It's almost like the C++ language committee doesn't want you to turn off exceptions! 😆 

The only solution that I can think of is to make the user provide two callbacks to `wait`, one to receive a value and one to receive an exception. At least then we could pass the complete, derived-class exception object to the exception callback. But clients would probably still need a way to figure out what type of exception was thrown in order to do anything useful with it. dynamic_cast? 😬 Plus that interface (two callbacks) is far less convenient.

So apps with exceptions disabled really just need to avoid rejected exceptions getting all the way to a wait. That's relatively easy to do just by adding a `catchImmediately` or `catchInMainThread` that turns a rejection into success with an error object.

In the future, if we stop wrapping Async++ and implement this stuff ourselves, we can probably do better. Not using exceptions to represent rejected future results, for example. But that's way outside the scope of this PR.

Reply:
yeah the alternative doesn't look good to me too.